src/main/java/com/olgaboiar/mint/Route.java

```diff
@@ -1,39 +1,19 @@
 package com.olgaboiar.mint;
 
-import java.util.Objects;
-
 public final class Route {
     private final String path;
-    private final String method;
+    String[] allowedMethods;
 
-    public Route(String path, String method) {
+    public Route(String path, String[] allowedMethods) {
         this.path = path;
-        this.method = method;
+        this.allowedMethods = allowedMethods;
     }
 
     public String getPath() {
         return path;
     }
 
-    public String getMethod() {
-        return method;
-    }
-
-    @Override
-    public boolean equals(Object route) {
-        if (route == this) {
-            return true;
-        }
-        if (!(route instanceof Route)) {
-            return false;
-        }
-        Route routeClone = (Route) route;
-        return Objects.equals(path, routeClone.path) &&
-                Objects.equals(method, routeClone.method);
-    }
-
-    @Override
-    public int hashCode() {
-        return Objects.hash(path, method);
+    public String[] getAllowedMethods() {
```

Comment:
Does the caller of this method need to access what's in the returned array? if so perhaps we can make it more explicit by using a map.